tinygrad/codegen/uopgraph.py

```diff
@@ -553,13 +557,22 @@ def get_recursive_children(x:UOp, end:UOps, include_self=False) -> Set[UOp]:
 
     # scope children impact the toposort and END* insertion
     scope_children = {p:get_recursive_children(p, END_FOR_UOP[p.op][0]) for p in reversed(in_degree) if p.op in END_FOR_UOP}
+    range_phi = {r:[p for p in scope_children[r] if p.op is UOps.PHI] for r in scope_children if r.op is UOps.RANGE}
+    range_srcs = {p:range_srcs[p] for p in range_srcs if p.op is UOps.PHI}
 
     queue:List[Tuple[int, UOp]] = []
     def push(u:UOp):
       priority = 0
-      # prefer uops that are loop children
-      for l, ss in scope_children.items():
-        if l.op is UOps.RANGE and u in ss: priority -= l.arg[0]*1000 + l.arg[1]
+      # ensure ranges fall in the proper order
+      if u.op is UOps.RANGE and u.arg[1]:
+        priority += u.arg[0]
+        # for each phi in this range's scope
```

Comment:
does this comment add value/context?

Reply:
not really, just writing out the thought process. I'll remove it in the next commit

Reply:
do you like the set().intersection change?

Reply:
Would using `any` be enough since range_phi is already deduped? Though .intersection is readable.
Fine with whichever one you like.

Reply:
yeah I can switch it back to any